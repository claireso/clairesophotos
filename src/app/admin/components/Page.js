import React from 'react'
import PropTypes from 'prop-types'

const Page = (props = {}) => {
  return (
    <main>
      <h1>{props.title}</h1>
      {props.children}
    </main>
  )
}

Page.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node
}

export default Page
