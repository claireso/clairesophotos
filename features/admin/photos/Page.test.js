import { render, waitFor } from '@testing-library/react'

import withTestRouter from '@utils/hoc/withTestRouter'

import Page from '../../../pages/admin/photos/index'

describe('Page photo', () => {
  const renderComponent = (routerProps) => {
    return render(
      withTestRouter(<Page />, {
        pathname: '/admin/photos',
        ...routerProps
      })
    )
  }

  beforeEach(() => {
    fetch.resetMocks()
    fetch.mockResponseOnce(JSON.stringify(global.__PHOTOS__))
  })

  test('should load and render photos', async () => {
    const { container, getByText } = renderComponent()

    expect(container).toMatchSnapshot()

    await waitFor(() => {
      expect(getByText('FÃ©vrier 2019')).toBeInTheDocument()
    })

    expect(container).toMatchSnapshot()
  })

  test('should display modal create', async () => {
    const { getByText } = renderComponent({
      query: {
        action: 'create_photo'
      }
    })

    await waitFor(() => {
      expect(getByText('Create a photo')).toBeInTheDocument()
    })
  })

  test('should display modal edit', async () => {
    const { getByText } = renderComponent({
      query: {
        action: 'edit_photo',
        id: 199
      }
    })

    await waitFor(() => {
      expect(getByText('Edit photo')).toBeInTheDocument()
    })
  })

  test('should display modal delete', async () => {
    const { getByText } = renderComponent({
      query: {
        action: 'delete_photo',
        id: 199
      }
    })

    await waitFor(() => {
      expect(getByText('Are you sure?')).toBeInTheDocument()
    })
  })
})
