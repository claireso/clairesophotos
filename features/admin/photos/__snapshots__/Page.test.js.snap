// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page photo should load and render photos 1`] = `
.c4 {
  margin: 2rem 0;
  -webkit-animation: jBcSpD 800ms ease-in-out infinite alternate;
  animation: jBcSpD 800ms ease-in-out infinite alternate;
}

.c4:after {
  background: var(--primary);
  border-radius: 50%;
  content: '';
  display: block;
  height: 1rem;
  width: 1rem;
  margin: 0 auto;
}

.c3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 0.2rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Roboto',Arial,sans-serif;
  font-size: 1.2rem;
  line-height: 1.3333333;
  outline: none;
  padding: 1.6rem;
  -webkit-transition: background 150ms ease-out;
  transition: background 150ms ease-out;
  background: var(--primary);
  color: white;
}

.c3 > svg {
  margin: 0 -0.4rem 0 0.4rem;
}

.c3 + .c2 {
  margin: 0 0 0 0.8rem;
}

.c3:hover {
  background: var(--primary-lighter);
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0 0 2rem;
}

.c0 > h1 {
  line-height: 1.835;
  margin: 0;
}

.c0 > h1 > span {
  color: var(--gray-2);
  font-size: 2rem;
  font-weight: normal;
  line-height: 1;
}

.c1 {
  font-size: 2.4rem;
  margin: 0 0 2.4rem;
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1"
    >
      Your photos 
      <span>
        (
        )
      </span>
    </h1>
    <button
      class="Buttonsstyles__Button-sc-4b7ll8-0 c2 c3"
    >
      Add a new photo
      <svg
        height="14"
        viewBox="0 0 24 24"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.984 12.984h-6v6h-1.97v-6h-6v-1.97h6v-6h1.97v6h6v1.97z"
        />
      </svg>
    </button>
  </div>
  <div
    class="c4"
  />
</div>
`;

exports[`Page photo should load and render photos 2`] = `
<div>
  <div
    class="Liststyles__ListHeader-sc-1ynnt8j-1 eBxLKn"
  >
    <h1
      class="Headingsstyles__Heading1-sc-6iuom-0 cfbPPp"
    >
      Your photos 
      <span>
        (
        184
        )
      </span>
    </h1>
    <button
      class="Buttonsstyles__Button-sc-4b7ll8-0 Buttonsstyles__StyledPrimaryButton-sc-4b7ll8-1 hrDGmN"
    >
      Add a new photo
      <svg
        height="14"
        viewBox="0 0 24 24"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.984 12.984h-6v6h-1.97v-6h-6v-1.97h6v-6h1.97v6h6v1.97z"
        />
      </svg>
    </button>
  </div>
  .c8 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 0.2rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Roboto',Arial,sans-serif;
  font-size: 1.2rem;
  line-height: 1.3333333;
  outline: none;
  padding: 1.6rem;
  -webkit-transition: background 150ms ease-out;
  transition: background 150ms ease-out;
  background: none;
  color: inherit;
  padding: 0.8rem;
}

.c8 > svg {
  margin: 0 -0.4rem 0 0.4rem;
}

.c8 + .c7 {
  margin: 0 0 0 0.8rem;
}

.c8 > svg {
  margin: 0;
}

.c8 + .c7 {
  margin: 0;
}

.c0 {
  margin: 0;
  padding: 0;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: grid;
  grid-template-columns: 8rem auto 12rem;
  grid-column-gap: 1rem;
  padding: 1.4rem;
  -webkit-transition: background 150ms ease-out;
  transition: background 150ms ease-out;
}

.c2:hover {
  background: var(--gray-5);
}

.c2 + .c1 {
  border-top: 1px solid var(--secondary);
}

.c3 {
  height: 8rem;
}

.c3 img {
  height: 100%;
  object-fit: contain;
  width: 100%;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c5 {
  color: var(--gray-3);
  font-size: 1.2rem;
}

.c6 {
  place-self: center flex-end;
  -webkit-transition: opacity 150ms ease-out;
  transition: opacity 150ms ease-out;
}

.c6 button {
  opacity: 0.5;
  -webkit-transition: opacity 150ms ease-out;
  transition: opacity 150ms ease-out;
}

.c6 button:hover {
  opacity: 1;
}

<ul
    class="c0"
  >
    <li
      class="c1 c2"
    >
      <div
        class="c3"
      />
      <div
        class="c4"
      >
        
        <p
          class="c5"
        >
          FÃ©vrier 2019
        </p>
      </div>
      <p
        class="c6"
      >
        <button
          class="Buttonsstyles__Button-sc-4b7ll8-0 c7 c8"
          title="Edit"
        >
          <svg
            height="16"
            version="1.1"
            viewBox="0 0 24 28"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.672 24l1.422-1.422-3.672-3.672-1.422 1.422v1.672h2v2h1.672zM13.844 9.5c0-0.203-0.141-0.344-0.344-0.344-0.094 0-0.187 0.031-0.266 0.109l-8.469 8.469c-0.078 0.078-0.109 0.172-0.109 0.266 0 0.203 0.141 0.344 0.344 0.344 0.094 0 0.187-0.031 0.266-0.109l8.469-8.469c0.078-0.078 0.109-0.172 0.109-0.266zM13 6.5l6.5 6.5-13 13h-6.5v-6.5zM23.672 8c0 0.531-0.219 1.047-0.578 1.406l-2.594 2.594-6.5-6.5 2.594-2.578c0.359-0.375 0.875-0.594 1.406-0.594s1.047 0.219 1.422 0.594l3.672 3.656c0.359 0.375 0.578 0.891 0.578 1.422z"
            />
          </svg>
        </button>
        <button
          class="Buttonsstyles__Button-sc-4b7ll8-0 c7 c8"
          title="Delete"
        >
          <svg
            height="16"
            version="1.1"
            viewBox="0 0 22 28"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 21.5v-11c0-0.281-0.219-0.5-0.5-0.5h-1c-0.281 0-0.5 0.219-0.5 0.5v11c0 0.281 0.219 0.5 0.5 0.5h1c0.281 0 0.5-0.219 0.5-0.5zM12 21.5v-11c0-0.281-0.219-0.5-0.5-0.5h-1c-0.281 0-0.5 0.219-0.5 0.5v11c0 0.281 0.219 0.5 0.5 0.5h1c0.281 0 0.5-0.219 0.5-0.5zM16 21.5v-11c0-0.281-0.219-0.5-0.5-0.5h-1c-0.281 0-0.5 0.219-0.5 0.5v11c0 0.281 0.219 0.5 0.5 0.5h1c0.281 0 0.5-0.219 0.5-0.5zM7.5 6h7l-0.75-1.828c-0.047-0.063-0.187-0.156-0.266-0.172h-4.953c-0.094 0.016-0.219 0.109-0.266 0.172zM22 6.5v1c0 0.281-0.219 0.5-0.5 0.5h-1.5v14.812c0 1.719-1.125 3.187-2.5 3.187h-13c-1.375 0-2.5-1.406-2.5-3.125v-14.875h-1.5c-0.281 0-0.5-0.219-0.5-0.5v-1c0-0.281 0.219-0.5 0.5-0.5h4.828l1.094-2.609c0.313-0.766 1.25-1.391 2.078-1.391h5c0.828 0 1.766 0.625 2.078 1.391l1.094 2.609h4.828c0.281 0 0.5 0.219 0.5 0.5z"
            />
          </svg>
        </button>
      </p>
    </li>
    <li
      class="c1 c2"
    >
      <div
        class="c3"
      />
      <div
        class="c4"
      >
        
        <p
          class="c5"
        >
          Janvier 2019
        </p>
      </div>
      <p
        class="c6"
      >
        <button
          class="Buttonsstyles__Button-sc-4b7ll8-0 c7 c8"
          title="Edit"
        >
          <svg
            height="16"
            version="1.1"
            viewBox="0 0 24 28"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.672 24l1.422-1.422-3.672-3.672-1.422 1.422v1.672h2v2h1.672zM13.844 9.5c0-0.203-0.141-0.344-0.344-0.344-0.094 0-0.187 0.031-0.266 0.109l-8.469 8.469c-0.078 0.078-0.109 0.172-0.109 0.266 0 0.203 0.141 0.344 0.344 0.344 0.094 0 0.187-0.031 0.266-0.109l8.469-8.469c0.078-0.078 0.109-0.172 0.109-0.266zM13 6.5l6.5 6.5-13 13h-6.5v-6.5zM23.672 8c0 0.531-0.219 1.047-0.578 1.406l-2.594 2.594-6.5-6.5 2.594-2.578c0.359-0.375 0.875-0.594 1.406-0.594s1.047 0.219 1.422 0.594l3.672 3.656c0.359 0.375 0.578 0.891 0.578 1.422z"
            />
          </svg>
        </button>
        <button
          class="Buttonsstyles__Button-sc-4b7ll8-0 c7 c8"
          title="Delete"
        >
          <svg
            height="16"
            version="1.1"
            viewBox="0 0 22 28"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 21.5v-11c0-0.281-0.219-0.5-0.5-0.5h-1c-0.281 0-0.5 0.219-0.5 0.5v11c0 0.281 0.219 0.5 0.5 0.5h1c0.281 0 0.5-0.219 0.5-0.5zM12 21.5v-11c0-0.281-0.219-0.5-0.5-0.5h-1c-0.281 0-0.5 0.219-0.5 0.5v11c0 0.281 0.219 0.5 0.5 0.5h1c0.281 0 0.5-0.219 0.5-0.5zM16 21.5v-11c0-0.281-0.219-0.5-0.5-0.5h-1c-0.281 0-0.5 0.219-0.5 0.5v11c0 0.281 0.219 0.5 0.5 0.5h1c0.281 0 0.5-0.219 0.5-0.5zM7.5 6h7l-0.75-1.828c-0.047-0.063-0.187-0.156-0.266-0.172h-4.953c-0.094 0.016-0.219 0.109-0.266 0.172zM22 6.5v1c0 0.281-0.219 0.5-0.5 0.5h-1.5v14.812c0 1.719-1.125 3.187-2.5 3.187h-13c-1.375 0-2.5-1.406-2.5-3.125v-14.875h-1.5c-0.281 0-0.5-0.219-0.5-0.5v-1c0-0.281 0.219-0.5 0.5-0.5h4.828l1.094-2.609c0.313-0.766 1.25-1.391 2.078-1.391h5c0.828 0 1.766 0.625 2.078 1.391l1.094 2.609h4.828c0.281 0 0.5 0.219 0.5 0.5z"
            />
          </svg>
        </button>
      </p>
    </li>
  </ul>
  .c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 0.2rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Roboto',Arial,sans-serif;
  font-size: 1.2rem;
  line-height: 1.3333333;
  outline: none;
  padding: 1.6rem;
  -webkit-transition: background 150ms ease-out;
  transition: background 150ms ease-out;
  background: var(--secondary);
  color: var(--text);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.4rem;
  height: 4.4rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  width: 4.4rem;
}

.c2 > svg {
  margin: 0 -0.4rem 0 0.4rem;
}

.c2 + .c1 {
  margin: 0 0 0 0.8rem;
}

.c2:hover {
  background: var(--secondary-darker);
}

.c2 + .c3 {
  margin-left: 0.8rem;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  list-style-type: none;
  width: 100%;
  padding: 0;
}

.c0 > li {
  margin: 0 0.4rem;
}

<ul
    class="c0"
  >
    <li>
      <button
        class="Buttonsstyles__Button-sc-4b7ll8-0 Buttonsstyles__SecondaryButton-sc-4b7ll8-3 c1 c2"
        title="Next page"
      >
        Â»
      </button>
    </li>
    <li>
      <button
        class="Buttonsstyles__Button-sc-4b7ll8-0 Buttonsstyles__SecondaryButton-sc-4b7ll8-3 c1 c2"
        title="Last page"
      >
        Â»Â»
      </button>
    </li>
  </ul>
</div>
`;
